File: backend-python/app/modules/system/dto/dept_request.py
from pydantic import BaseModel, Field


class DeptSaveRequest(BaseModel):
    parent_id: int | None = Field(default=None, alias="parentId")
    dept_name: str = Field(alias="deptName", min_length=1, max_length=64)
    dept_code: str = Field(alias="deptCode", min_length=1, max_length=64)
    leader_user_id: int | None = Field(default=None, alias="leaderUserId")
    sort: int | None = None
    status: int | None = Field(default=None, ge=0, le=1)

File: backend-rust/src/modules/system/dto/dept_request.rs
use serde::Deserialize;

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct DeptSaveRequest {
    pub parent_id: Option<i64>,
    pub dept_name: String,
    pub dept_code: String,
    pub leader_user_id: Option<i64>,
    pub sort: Option<i32>,
    pub status: Option<i32>,
}
